Taxi/Taxi/Presenter/SignUpNicknameView.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  SignUpNicknameView.swift
+//  Taxi
+//
+//  Created by sanghyo on 2022/06/14.
+//
+
+import SwiftUI
+
+struct SignUpNicknameView: View {
+    @EnvironmentObject var userViewModel: Authentication
+    @Environment(\.dismiss) private var dismiss
+    @State private var nickname: String = ""
+    @State private var nickFieldState: FieldState = .normal
+    @FocusState private var focusField: Bool
+    private let deviceUUID = UIDevice.current.identifierForVendor!.uuidString
+
+    var body: some View {
+        VStack {
+            VStack(alignment: .leading, spacing: 80) {
+                Text("닉네임을 입력해주세요")
+                    .signUpTitle()
+                UnderlinedTextField(text: $nickname, nickFieldState, "닉네임")
+                    .font(Font.custom("AppleSDGothicNeo-Regular", size: 20))
+                    .focused($focusField)
+            }
+            .padding(.horizontal)
+            makeMessage(nickname: nickname)
+                .font(Font.custom("AppleSDGothicNeo-Regular", size: 14))
+                .foregroundColor(setMessageColor(nickname: nickname))
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .padding([.top, .horizontal])
+            Spacer()
+            makeConditionalButton("완료", !String.isValidNickname(nickname: nickname).isValid, focusField) {
+                print("완료버튼 눌림")
+                userViewModel.register(id: deviceUUID, nickname: nickname)
+                // TODO: 유저 환영 화면 연결 or pop to root
+                UserDefaults.standard.set(true, forKey: "isLogined")
+            }
+        }
+        .navigationBarBackButtonHidden(true)
+        .toolbar {
+            ToolbarItem(placement: .navigationBarLeading) {
+                Button {
+                    dismiss()
+                } label: {
+                    Image(systemName: "chevron.backward")
+                }
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func makeConditionalButton(_ title: String, _ disabled: Bool = false, _ focusState: Bool, action: @escaping () -> Void) -> some View {
+        if focusState {
+            FlatButton(title, disabled, action: action)
+        } else {
+            RoundedButton(title, disabled, action: action)
+        }
+    }
+
+    @ViewBuilder
+    private func makeMessage(nickname: String) -> some View {
```

Comment:
```suggestion
    private func makeMessageView(nickname: String) -> some View {
```

Message를 만드는 메소드인지 MessageView를 만드는 메소드인지의 구분이 필요해보여요

Reply:
감사합니당 어떻게 그냥 메소드와 뷰를 만드는 메소드를 구분할 수 있을지 고민이었는데 도움이 됐습니다 ㅎㅎ